### ISSUE TITLE
planefinding.dll and spatialunderstanding.dll causing validation errors in WACK

### ISSUE THREAD

**Issue by lukasvolf** [Microsoft/MixedRealityToolkit-Unity#746]
We are trying to submit our UWP app to the Store, but we are getting WACK errors regarding unsupported APIs:
For x64, we are getting the following:
```
API FindPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API FindSubPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API MergeSubPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.

For ARM, we are getting even more errors:
API ActivateShapeAnalysis in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API AddShape in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API GeneratePlayspace_ExtractMesh_Extract in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API GeneratePlayspace_ExtractMesh_Setup in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API GeneratePlayspace_InitScan in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API GeneratePlayspace_RequestFinish in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API GeneratePlayspace_UpdateScan in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API PlayspaceRaycast in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryPlayspaceAlignment in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryPlayspaceStats in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryShape_FindPositionsOnShape in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryShape_FindShapeHalfDims in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindLargePositionsOnWalls in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindLargestPositionsOnFloor in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindLargestWall in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindPositionsOnFloor in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindPositionsOnWalls in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API QueryTopology_FindPositionsSittable in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API RemoveAllShapes in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API Solver_Init in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API Solver_PlaceObject in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API Solver_RemoveAllObjects in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API Solver_RemoveObject in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API SpatialUnderstanding_Init in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API SpatialUnderstanding_Term in spatialunderstanding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API FindPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API FindSubPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
API MergeSubPlanes in planefinding.dll is not supported for this application type. Assembly-CSharp.dll calls this API.
```

x86 passes the WACK validation with no errors.

We are using HoloToolkit-Unity-1.5.6.0. What should we do to fix this? The app is not HoloLens-only, so we need to support all platforms.

**Comment by StephenHodgson** [comment]
Short answer, just remove the dlls from the cross platform app versions. I highly doubt those devices are calling those dlls, but if they are, maybe stub out some empty methods instead.

Really, this would need to be fixed in the main HoloToolkit, and recompiled.

**Comment by lukasvolf** [comment]
Deleting those DLLs did not help, the errors are the same. I think the issue is in the HoloToolkit itself where those functions are being imported and called. The easiest solution that I can think of is to define x64/x86 and ARM directives in csc.rsp and compile-switch those imports and functions calling them. We determine whether we are running on HoloLens in the runtime so the code is not being called on those platforms anyway.

**Comment by keveleigh** [comment]
This should have been fixed a while ago via #1670, which added all variations of the DLLs.
